node/src/blockchain/blockchain_interface.rs

```diff
@@ -344,7 +344,7 @@ where
             )
             .build();
 
-        let fallback_end_block_number = match end_block {
+        let fallback_start_block_number = match end_block {
             BlockNumber::Number(eb) => eb.as_u64(),
             _ => {
                 if let BlockNumber::Number(start_block_number) = start_block {
```

Comment:
I can't put this comment any lower, but it should be on the `panic!()` below. Can you expand and elaborate this error message? If I got it, I wouldn't have any idea what was wrong.

Reply:
This is the only supported usage of the enum BlockNumber `BlockNumber::Number(eb)`. The other 3 enum values are useless and unsupported here: `BlockNumber::Latest`, `BlockNumber::Earliest`, `BlockNumber::Pending`. This is a `panic!()` intended for the developer who calls `retrieve_transactions()` and passes in one of the unsupported values of the enum. It has to be a real block number. There's an earlier initial call to find out what the Latest block number is so it can be passed to this method. You may remember that originally I had the parameter `start_block` defined as `u64` however it seemed too strange and unexpected to have `start_block: u64` while `end_block: BlockNumber` so I defined them the same but protected from using the wrong one.

Should I change the message to something like:
`panic!("Developer: don't pass start_block as BlockNumber::Latest, BlockNumber::Earliest, or BlockNumber::Pending; they are not supported");
?